tinycrypt/ecc.c

```diff
@@ -203,14 +210,177 @@ int uECC_curve_public_key_size(void)
 	return 2 * NUM_ECC_BYTES;
 }
 
+#if defined MBEDTLS_OPTIMIZE_TINYCRYPT_ASM && defined __CC_ARM
+__asm void uECC_vli_clear(uECC_word_t *vli)
+{
+#if NUM_ECC_WORDS != 8
+#error adjust ARM assembly to handle NUM_ECC_WORDS != 8
+#endif
+#if !defined __thumb__ || __TARGET_ARCH_THUMB < 4
+    MOVS    r1,#0
+    MOVS    r2,#0
+    STMIA   r0!,{r1,r2}
+    STMIA   r0!,{r1,r2}
+    STMIA   r0!,{r1,r2}
+    STMIA   r0!,{r1,r2}
+    BX      lr
+#else
+    MOVS    r1,#0
+    STRD    r1,r1,[r0,#0]       // Only Thumb2 STRD can store same reg twice, not ARM
+    STRD    r1,r1,[r0,#8]
+    STRD    r1,r1,[r0,#16]
+    STRD    r1,r1,[r0,#24]
+    BX      lr
+#endif
+}
+#elif defined MBEDTLS_OPTIMIZE_TINYCRYPT_ASM && defined __GNUC__ && defined __arm__
```

Comment:
@AndrzejKurek , @kjbracey-arm , as the flags defined right now , it seems this #elif is never stepped in . In baremetal.h defined 

```
#if !( defined(__STRICT_ANSI__) && defined(__CC_ARM) )
    #define MBEDTLS_OPTIMIZE_TINYCRYPT_ASM
```
and in this file 
```

#if defined MBEDTLS_OPTIMIZE_TINYCRYPT_ASM && defined __CC_ARM
.
.
.
#elif defined MBEDTLS_OPTIMIZE_TINYCRYPT_ASM && defined __GNUC__ && defined __arm__
```
What are we trying to achieve here ?

Reply:
Understood it